### PR TITLE
Local folder content summary with ChatGPT to Microsoft SQL table

### LINKED ISSUE #110: Summarize file contents using ChatGPT and save to database
1. Read files from local disk or from SFTP (whichever one is simpler)
2. For each file - ask the ChatGPT to summarize the contents of the file.
3. Save results of summaries to the database. The database table should include file name, timestamp when analysis was done and summary.

Parameters:
1. Source file inputs
2. ChatGPT creds
3. ChatGPT model to use
4. Prompt for summarization (lets have a meaningful default)
5. SQL DB creds / table name

We have a task for ChatGPT

### PR DESCRIPTION
**Pull request by FrendsSarlinS**
Closes #110 

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters.
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots.
- [x] Template parameters checked
  - [x] Parameter descriptions are good and end with dots.
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output
- [x] End "balls" like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Tempalte files are going to the correct folder

## Summary by CodeRabbit

- **New Features**
  - Introduced a template that reads local folder contents, uses ChatGPT to summarize each file, and inserts summaries into a Microsoft SQL database table.
  - Added metadata and process templates to support the summarization and database insertion workflow.
  - Included error handling and logging for improved reliability.

- **Documentation**
  - Added detailed documentation on prerequisites, implementation details, and usage notes for the new template.

### CODE REVIEW

**Review by jannejjj — COMMENTED** [review]
- The process variables don't use all caps for "GPT", e.g., "ChatGptModel", it's a small thing but i think it should be capitalized 😄

- The final return could probably state the total number of files, which I presume can be gotten from the find files task.

**Comment by FrendsSarlinS** [comment]
Thanks for the comment, Janne! The changes you suggested have been implemented.

I also parametrized the values in MicrosoftSQL ExecuteQuery -task, because a single ' in the response could result in an exception.

**Review by jannejjj — APPROVED** [review]
Changes have been made according to the comments, and also good job spotting that the summary should also be parametrized to avoid issues with quotes!

File: Templates/Files/Local Files/Local folder content summary with ChatGPT to Microsoft SQL table/long-description.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+This template assumes that the following prerequisites are in place:
+
+- A Microsoft SQL database and a database table with appropriate columns exist for inserting the summary.
+- ChatGPT API key is accessible.
+
+# Implementation and Usage Notes
+
+This template finds a local folder and checks the folder for files. For each file in the local folder, the file content is read and submitted to ChatGPT with a prompt to summarize the file content.
+The summary is inserted to the Microsoft SQL database table with an SQL query. The task for executing the query utilizes parameters to prevent SQL injections. The database should contain columns for at least the file name, the date and time the summary is input to the table, and the summary itself.
```
Comment (jefim) [inline Templates/Files/Local Files/Local folder content summary with ChatGPT to Microsoft SQL table/long-description.md:16]: Please provide the sql create table for the target table.

File: Templates/Files/Local Files/Local folder content summary with ChatGPT to Microsoft SQL table/long-description.md
Comment (jefim) [inline Templates/Files/Local Files/Local folder content summary with ChatGPT to Microsoft SQL table/long-description.md:1]: Markdown requires two newlines to start a new paragraph (= one empty line inbetween paragraphs), so some newlines could be useful here to separate the text into paragraphs.

**Review by jefim — CHANGES_REQUESTED** [review]

**Review by jannejjj — APPROVED** [review]
The description has been adjusted according to the review left by Jefim.

**Pull request merged**